lib/UniversalPeripheral.dart

```diff
@@ -23,6 +23,15 @@ class UniversalPeripheral {
     _protocol.init();
   }
 
+  // returns extened FEN, which indicate board state
+  // for devices, which not supports piece recognition
+  // (e.g. based on hall sensors) FEN can contain "?" characteres
+  // indicating only piece occurance
+  // returns null when board is not supporting preview
+  String getState() {
```

Comment:
We could think about throwing a `MethodNotImplementedException` instead.
Just an idea but this would make the function typesafe if we upgrade to flutter null safety.

Reply:
Great:

changed to:
```
  String getState() {
    throw new UnimplementedError("preview not supported");
  }
```